tools/gen-intrinsics-specs.py

```diff
@@ -93,51 +94,86 @@ def rst_literal_quote(mapping, workflow):
+    >>> quote_split_intrinsics('int f(int x, float y)', 'markdown', 'baseurl/')
+    '<code>int <a href="baseurl/f" target="_blank">f</a>(<br>&nbsp;&nbsp;&nbsp;&nbsp; int x,<br>&nbsp;&nbsp;&nbsp;&nbsp; float y)</code>'
 
-    >>> quote_split_intrinsics('int f(int x)', 'markdown')
-    '`int f(int x)`'
+    >>> quote_split_intrinsics('int f(int x)', 'markdown', 'baseurl/')
+    '<code>int <a href="baseurl/f" target="_blank">f</a>(int x)</code>'
 
-    >>> quote_split_intrinsics('int f(int x, float y)', 'pdf')
+    >>> quote_split_intrinsics('int f(int x, float y)', 'pdf', 'baseurl/')
     '``` c\nint f(\n  int x,\n  float y)\n```'
 
-    >>> quote_split_intrinsics('int f(int x)', 'pdf')
+    >>> quote_split_intrinsics('int f(int x)', 'pdf', 'baseurl/')
     '`int f(int x)`'
     """
-    # Remove the suffix ')' from the intrinsic string.
-    intrinsic_without_ending = intrinsic[:-1]
-    ret_def, par, signature = intrinsic_without_ending.partition('(')
-    split_signature = signature.split(',')
+
     whitespace_indent = "&nbsp;&nbsp;&nbsp;&nbsp;"
 
-    if len(split_signature) > 1:
+    intrinsic_signature = get_signature(intrinsic)
+    intrinsic_link_id = get_intrinsic_name(intrinsic)
+    intrinsic_type = get_intrinsic_return_type(intrinsic)
+
+    # This replacement has been added in order to avoid issues with links
+    # that contain square brackets.
+    # See also: https://stackoverflow.com/questions/13013987/ruby-how-to-escape-url-with-square-brackets-and/17901435
+    formatted_intrinsic_id = intrinsic_link_id.replace("[","%5B").replace("]","%5D")
+    formatted_site_link = f"{baseurl}{formatted_intrinsic_id}"
+
+    if len(intrinsic_signature) > 1:
         if workflow == "rst":
-            return f".. code:: c\n\n    {ret_def}(\n        " + ',\n       '.join(split_signature) + ")"
+            return f".. code:: c\n\n    {intrinsic_type} {intrinsic_link_id}(\n        " + ',\n       '.join(intrinsic_signature) + ")"
         elif workflow == "markdown":
-            return f"`{ret_def}(`<br>" + whitespace_indent + "`" + (',`<br>'+ whitespace_indent + '`').join(split_signature) + ")`"
+            return f"<code>{intrinsic_type} <a href=\"{formatted_site_link}\" target=\"_blank\">{intrinsic_link_id}</a>(<br>{whitespace_indent} " + (f",<br>{whitespace_indent}").join(intrinsic_signature) + ")</code>"
         elif workflow == "pdf":
-            return f"``` c\n{ret_def}(\n  " + (',\n ').join(split_signature) + ")\n```"
+            return f"``` c\n{intrinsic_type} {intrinsic_link_id}(\n  " + (',\n ').join(intrinsic_signature) + ")\n```"
     else:
         if workflow == "rst":
             return f".. code:: c\n\n    {intrinsic}\n"
         elif workflow == "markdown":
-            return f"`{intrinsic}`"
+            return f"<code>{intrinsic_type} <a href=\"{formatted_site_link}\" target=\"_blank\">{intrinsic_link_id}</a>({intrinsic_signature[0]})</code>"
```

Comment:
Same here, please split this line. And also, add an assert that check that `len(intrinsic_signature) == 1`, otherwise the code will fail if we ever add an intrinsic that does not have input parameters.